app/selectors/post_list.js

```diff
@@ -42,6 +43,10 @@ export function makePreparePostIdsForPostList() {
             for (let i = posts.length - 1; i >= 0; i--) {
                 const post = posts[i];
 
+                if (post.type === Posts.POST_TYPES.EPHEMERAL_ADD_TO_CHANNEL && !selectedPostId) {
```

Comment:
I don't understand this. Why are we not showing EPHEMERAL_ADD_TO_CHANNEL posts when there's no selected post ID?

Reply:
Maybe I should add a comment on this logic (let me know if needed).
The way to identify active screen is: Thread screen is active if `selectedPostId` is not empty, and then any other screen like Channel if it's empty.
When we added a user at Thread screen, we are posting ephemeral post that the user is added on Thread screen itself. However we are not posting that same ephemeral post at Channel screen since there is a similar system post coming from the server that the user is added. If it's not filtered out, there will be 2 posts at Channel screen.

Reply:
Maybe we could change it on the server then so that it replies to the thread if you're at mentioning someone in a reply? Having the message sometimes created by the server and sometimes by the client seems error prone